corehq/apps/hqcase/views.py

```diff
@@ -62,3 +70,50 @@ def delete_cases(self, request, domain):
         res = delete_exploded_case_task.delay(self.domain, explosion_id)
         download.set_task(res)
         return redirect('hq_soil_download', self.domain, download.download_id)
+
+
+# TODO switch to @require_can_edit_data
+@waf_allow('XSS_BODY')
+@csrf_exempt
+@api_auth
+@require_superuser_or_contractor
+@requires_privilege_with_fallback(privileges.API_ACCESS)
```

Comment:
@esoergel I was playing with this API a bit today and realized that it does not have CORS support or Oauth support. I think these things would be a powerful addition to allow external applications to modify case data via the front end.

I'm curious if this was an intentional design decision, an oversight, or just something that will come later when the API is relased?

Reply:
I agree that both of those would be a valuable addition.  I thought you were still actively working on OAuth, and was imagining slapping on an appropriate decorator at some point when that work was done - do you know where we are in regards to that?  Is there work to be done to use OAuth in a non tastypie view?

I hadn't looked into supporting CORS, but that does also seem appropriate - just added it to the "TODOs".  Let me know if you have other such feedback on the API - next steps are to have QA take a pass at it, then I want to get feedback from some partners who might use it to see what else it needs to meet their use-cases before releasing it more generally.  Feel free to add a section in the doc too if you've got notes:
https://docs.google.com/document/d/1upMFrcItnyxVkD2NoBzpO6pTJRMncYVRVQcrvF5Gfj0/edit#

Reply:
Awesome, glad to hear you're supportive.

I actually got both oauth and CORS working locally yesterday so I'll try and clean them up and submit some PRs soon.

And I'll definitely update that doc if I have any feedback from using the API.

Reply:
https://github.com/dimagi/commcare-hq/pull/29367